### ISSUE TITLE
<MAXSTRING> when set float to gref

### ISSUE THREAD

**Issue by daimor** [intersystems-community/embedded-python-bugreports#10]
**Describe the bug**
set any float value to global causes `<MAXSTRING>` error

**iris version/image**
IRIS for UNIX (Ubuntu Server LTS for ARM64 Containers) 2023.2 (Build 227U) Mon Jul 31 2023 17:43:25 EDT

**python version**
any

**To Reproduce**
```
USER>:py

Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type quit() or Ctrl-D to exit this shell.
>>> iris.gref("^tmp")
<iris.gref object at 0xfffddc541a70>
>>> iris.gref("^tmp")['test'] = ""
>>> iris.gref("^tmp")['test'] = 0
>>> iris.gref("^tmp")['test'] = 0.0
Traceback (most recent call last):
  File "<input>", line 1, in <module>
RuntimeError: IRIS error <MAXSTRING>
>>> iris.gref("^tmp").set(['test'], 0.0)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
RuntimeError: IRIS error <MAXSTRING>
```

**Comment by grongierisc** [comment]
I confirm this bug, it only occur on 2023.2.

A fix is on it's way, should be fixed on 2023.3.

Thanks for reporting it.

**Comment by daimor** [comment]
kind of solved in 2023.3, see #13
